### PR TITLE
New serverless pattern - cdk-sfn-s3

### PR DESCRIPTION
**Pull request by Harshi719**
*Issue #, if available:*

*Description of changes:* New pattern cdk-sfn-s3 is added along with readme and example-pattern.json file

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by boyney123** [comment]
Hey @Harshi719 

Thanks for the contribution.

I see you have your `.idea` folder committed into this PR. These are configuration files setup in your IDE (webstorm I think), could you remove them please? 

Thanks!

**Comment by Harshi719** [comment]
@boyney123 [boyney123](https://github.com/boyney123)  Thanks for bringing into my notice , have 
removed .idea folder

File: cdk-sfn-s3/bin/cdk-sfn-s3.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env node
+import 'source-map-support/register';
+import * as cdk from '@aws-cdk/core';
+import { CdkSfnS3Stack } from '../lib/cdk-sfn-s3-stack';
```
Comment (boyney123) [inline cdk-sfn-s3/bin/cdk-sfn-s3.ts:4]: Hey,

Is this file missing in the PR? I cant find the `cdk-sfn-s3-stack` file anywhere?

File: cdk-sfn-s3/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+     ```
+ 6. Deploy the stack to your default AWS account and region. The output of this command should give you the StepFunctions State Machine ARN.
+     ```
+     cdk deploy
+     ```
+ 8. Note the outputs from the CDK deployment process. These contain the ARN of StepFunctions State Machine.
+
+## How it works
+
+The AWS CDK  template deploys a Step Functions workflow that publishes an event to an EventBridge event bus.
```
[the remark this answers is not in the document]
Comment (Harshi719) [reply cdk-sfn-s3/README.md]: I have corrected and checked in

File: cdk-sfn-s3/bin/cdk-sfn-s3.ts (hunk 1 above)
Comment (Harshi719) [reply cdk-sfn-s3/bin/cdk-sfn-s3.ts:4]: add the missing lib folder , somehow it was with .gitignore. Thanks

File: cdk-sfn-s3/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (boyney123) [reply cdk-sfn-s3/README.md]: Thanks!

**Comment by boyney123** [comment]
Thanks for the contribution @Harshi719 , just reviewed and tested looks good to me.

**Pull request merged**